src/compiler/objective_c_generator_helpers.h

```diff
@@ -36,10 +36,29 @@ inline string MessageHeaderName(const FileDescriptor* file) {
   return google::protobuf::compiler::objectivec::FilePath(file) + ".pbobjc.h";
 }
 
+inline bool AsciiIsUpper(char c) { return c >= 'A' && c <= 'Z'; }
+
 inline string ServiceClassName(const ServiceDescriptor* service) {
   const FileDescriptor* file = service->file();
-  string prefix = google::protobuf::compiler::objectivec::FileClassPrefix(file);
-  return prefix + service->name();
+  string prefix = proto2::compiler::objectivec::FileClassPrefix(file);
```

Comment:
will still need to access the method via google::protobuf::compiler::objectivec namespace since this is from the OSS package. 

```
In file included from src/compiler/objective_c_plugin.cc:27:
./src/compiler/objective_c_generator_helpers.h:43:19: error: use of undeclared identifier 'proto2'
  string prefix = proto2::compiler::objectivec::FileClassPrefix(file);
                  ^
1 error generated.
```
```

Reply:
Fixed. Apologies for the rookie error :(